File: CSharp/ParrotTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace parrot
{
    [TestClass]
    public class ParrotTest
    {
        [TestMethod]
        public void GetSpeedOfEuropeanParrot()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.EUROPEAN, 0, 0, false);
            Assert.AreEqual(12.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedOfAfricanParrot_With_One_Coconut()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.AFRICAN, 1, 0, false);
            Assert.AreEqual(3.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedOfAfricanParrot_With_Two_Coconuts()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.AFRICAN, 2, 0, false);
            Assert.AreEqual(0.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedOfAfricanParrot_With_No_Coconuts()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.AFRICAN, 0, 0, false);
            Assert.AreEqual(12.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedNorwegianBlueParrot_nailed()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 0, true);
            Assert.AreEqual(0.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedNorwegianBlueParrot_not_nailed()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 1.5, false);
            Assert.AreEqual(18.0, parrot.GetSpeed());
        }

        [TestMethod]
        public void GetSpeedNorwegianBlueParrot_not_nailed_high_voltage()
        {
            Parrot parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 4, false);
            Assert.AreEqual(24.0, parrot.GetSpeed());
        }
    }
}

File: C/ParrotTest.c
#include <setjmp.h>
#include <stdarg.h>
#include <stdlib.h>

#include <cmocka.h>

#include "Parrot.h"

static void get_speed_of_european_parrot(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, EUROPEAN, 0, 0, false);
    assert_float_equal(12.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_african_parrot_with_one_coconut(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, AFRICAN, 1, 0, false);
    assert_float_equal(3.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_african_parrot_with_two_coconuts(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, AFRICAN, 2, 0, false);
    assert_float_equal(0.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_african_parrot_with_no_coconuts(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, AFRICAN, 0, 0, false);
    assert_float_equal(12.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_norwegian_blue_parrot_nailed(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, NORWEGIAN_BLUE, 0, 1.5, true);
    assert_float_equal(0.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_norwegian_blue_parrot_not_nailed(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, NORWEGIAN_BLUE, 0, 1.5, false);
    assert_float_equal(18.0, get_speed(&parrot), 0.0);
}

static void get_speed_of_norwegian_blue_parrot_not_nailed_high_voltage(void** state)
{
    (void)state; /* unused */
    Parrot parrot;
    init_parrot(&parrot, NORWEGIAN_BLUE, 0, 4, false);
    assert_float_equal(24.0, get_speed(&parrot), 0.0);
}

int main(void)
{
    const struct CMUnitTest test_suite[] = {
        cmocka_unit_test(get_speed_of_european_parrot),                  /* */
        cmocka_unit_test(get_speed_of_african_parrot_with_one_coconut),  /* */
        cmocka_unit_test(get_speed_of_african_parrot_with_two_coconuts), /* */
        cmocka_unit_test(get_speed_of_african_parrot_with_no_coconuts),  /* */
        cmocka_unit_test(get_speed_of_norwegian_blue_parrot_nailed),     /* */
        cmocka_unit_test(get_speed_of_norwegian_blue_parrot_not_nailed), /* */
        cmocka_unit_test(get_speed_of_norwegian_blue_parrot_not_nailed_high_voltage), /* */
    };

    return cmocka_run_group_tests(test_suite, NULL, NULL);
}
